src/game/game/game.c

```diff
@@ -207,7 +216,9 @@ int32_t Game_Stop(void)
     if (g_CurrentLevel == g_GameFlow.last_level_num) {
         g_Config.profile.new_game_plus_unlock = true;
         Config_Write();
-    } else {
+    }
+
+    if (g_CurrentLevel < g_GameFlow.level_count) {
```

Comment:
I understand that this got removed to carry over save information to the bonus levels, and this works because the bonus levels usually happen to be placed after the normal levels.

What happens if the game doesn't have any bonus levels though and we're on the last level? I think there will be a memory corruption and to avoid it, this should become `g_CurrentLevel + 1 < g_GameFlow.level_count`.